### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by rclee**
We need to replace `didRecieveChallenge` related test since we moved directions. Comparing the received disposition from the handler is more valuable. 

ref: https://github.com/mapbox/mapbox-events-ios/pull/74

### CODE REVIEW

File: MapboxMobileEventsTests/MMEAPIClientTests.mm [hunk 1]
```diff
@@ -31,6 +31,9 @@ - (void)setupUserAgent;
     
     __block MMEAPIClient *apiClient;
     __block int64_t timeoutInNanoseconds = 1000000000;
+    __block NSURLSessionAuthChallengeDisposition receivedDisposition;
+    __block NSURLSessionAuthChallengeDisposition expectedDefaultDisposition = NSURLSessionAuthChallengePerformDefaultHandling;
```
Comment (akitchen) [inline MapboxMobileEventsTests/MMEAPIClientTests.mm]: `__block` is only needed for references which will be assigned to (changed) within a block other than the current scope. `timeoutInNanoseconds`, `expectedDefaultDisposition`, and `expectedCancelDisposition` technically don't require `__block`

File: MapboxMobileEventsTests/MMEAPIClientTests.mm [hunk 2]
```diff
@@ -84,19 +87,15 @@ - (void)setupUserAgent;
...
             });
             
-            it(@"should receive challenge", ^{
-                apiClient.sessionWrapper should have_received(@selector(URLSession:didReceiveChallenge:completionHandler:));
-            });
-            
-            it(@"should not receive challenge", ^{
-                apiClient.sessionWrapper should_not have_received(@selector(URLSession:task:didReceiveChallenge:completionHandler:));
+            it(@"should equal expected CancelAuthentication disposition", ^{
+                expectedCancelDisposition should equal(receivedDisposition);
```
Comment (akitchen) [inline MapboxMobileEventsTests/MMEAPIClientTests.mm]: I would write this assertion the other way around

Comment (akitchen) [reply MapboxMobileEventsTests/MMEAPIClientTests.mm]: same thing below...

**Comment by akitchen** [comment]
I think this is a good improvement, but ultimately we care about testing the results of 

https://github.com/mapbox/mapbox-events-ios/blob/5075b42512ca8675a77b8fd85f0f629eb22c93c3/MapboxMobileEvents/MMENSURLSessionWrapper.m#L73

which, to do right (and to eliminate the possible need for hitting the network during test runs) we should be injecting a fake `trustKit.pinningValidator` which we can use to simulate/control the results of `-[pinningValidator handleChallenge:completionHandler:]` in the tests

**Review by akitchen — DISMISSED** [review]

**Review by alfwatt — APPROVED** [review]
Implemented review comments

**Pull request merged**